lib/plugins/aws/package/compile/layers/index.js

```diff
@@ -51,56 +53,127 @@ class AwsCompileLayers {
+      const newLayerHashOutput = this.cfOutputLayerHashTemplate();
+      newLayerHashOutput.Value = sha;
+
+      const layerS3KeyOutputLogicalId = this.provider.naming.getLambdaLayerS3KeyOutputLogicalId(
+        layerName
+      );
+      const newLayerS3KeyOutput = this.cfOutputLayerS3KeyTemplate();
+      newLayerS3KeyOutput.Value = newLayer.Properties.Content.S3Key;
+
+      _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Outputs, {
+        [layerOutputLogicalId]: newLayerOutput,
+        [layerHashOutputLogicalId]: newLayerHashOutput,
+        [layerS3KeyOutputLogicalId]: newLayerS3KeyOutput,
       });
-    }
+    });
+  }
 
-    _.merge(
-      this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
-      newLayerObject
+  compareWithLastLayer(layerName) {
+    const stackName = this.provider.naming.getStackName();
+    const layerHashOutputLogicalId = this.provider.naming.getLambdaLayerHashOutputLogicalId(
+      layerName
     );
 
-    // Add layer to Outputs section
-    const layerOutputLogicalId = this.provider.naming.getLambdaLayerOutputLogicalId(layerName);
-    const newLayerOutput = this.cfOutputLayerTemplate();
-
-    newLayerOutput.Value = { Ref: layerLogicalId };
+    return this.provider.request('CloudFormation', 'describeStacks', { StackName: stackName }).then(
+      data => {
+        const lastHash = data.Stacks[0].Outputs.find(
+          output => output.OutputKey === layerHashOutputLogicalId
+        );
+        const compiledCloudFormationTemplate = this.serverless.service.provider
+          .compiledCloudFormationTemplate;
+        const newSha = compiledCloudFormationTemplate.Outputs[layerHashOutputLogicalId].Value;
+        if (lastHash == null || lastHash.OutputValue !== newSha) {
+          return;
+        }
 
-    _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Outputs, {
-      [layerOutputLogicalId]: newLayerOutput,
-    });
+        const layerS3keyOutputLogicalId = this.provider.naming.getLambdaLayerS3KeyOutputLogicalId(
+          layerName
+        );
+        const lastS3Key = data.Stacks[0].Outputs.find(
+          output => output.OutputKey === layerS3keyOutputLogicalId
+        );
+        compiledCloudFormationTemplate.Outputs[layerS3keyOutputLogicalId].Value =
+          lastS3Key.OutputValue;
+        const layerLogicalId = this.provider.naming.getLambdaLayerLogicalId(layerName);
+        const layerResource =
+          compiledCloudFormationTemplate.Resources[layerLogicalId] ||
+          compiledCloudFormationTemplate.Resources[`${layerLogicalId}${lastHash.OutputValue}`];
+        layerResource.Properties.Content.S3Key = lastS3Key.OutputValue;
+        const layerObject = this.serverless.service.getLayer(layerName);
+        layerObject.artifactAlreadyUploaded = true;
+        this.serverless.cli.log(`Layer ${layerName} is already uploaded.`);
+      },
+      e => {
+        if (e.message.indexOf('does not exist') > -1) {
```

Comment:
Let's rely on `str.includes` instead

Reply:
Fixed in https://github.com/serverless/serverless/pull/7680/commits/a44ced8e89bde9874ab39b3bcd2200573591852c